cla_backend/apps/cla_butler/tests/test_command.py

```diff
@@ -19,3 +21,50 @@ def delay(*args, **kwargs):
         mock_delay.side_effect = delay
         management.call_command("diversity_data_check")
         self.assertDictEqual(delay_result, expected_delay_result)
+
+
+class DiversityReencryptCommandTestCase(CreateSampleDiversityData, TestCase):
+    @mock.patch(
+        "cla_butler.management.commands.diversity_data_reencrypt.Command.get_old_key_passphrase", return_value="cla"
+    )
+    @mock.patch("cla_butler.management.commands.diversity_data_reencrypt.Command.schedule_tasks")
+    @mock.patch.dict("os.environ", {"PREVIOUS_DIVERSITY_PRIVATE_KEY": "I am not empty"})
+    def test_create_tasks_all(self, mock_schedule_tasks, get_old_key_passphrase):
+        ClaButlerDiversityReencryptCommand.chunk_size = 2
+
+        def schedule_tasks(tasks, passphrase_old, **kwargs):
+            tasks_count = len(self.pd_records_ids) / ClaButlerDiversityReencryptCommand.chunk_size
+            self.assertEqual(len(tasks), tasks_count)
+
+        mock_schedule_tasks.side_effect = schedule_tasks
+        management.call_command("diversity_data_reencrypt")
+        self.assertTrue(mock_schedule_tasks.called)
+
+    @mock.patch(
+        "cla_butler.management.commands.diversity_data_reencrypt.Command.get_old_key_passphrase", return_value="cla"
+    )
+    @mock.patch("cla_butler.management.commands.diversity_data_reencrypt.Command.schedule_tasks")
+    @mock.patch.dict("os.environ", {"PREVIOUS_DIVERSITY_PRIVATE_KEY": "I am not empty"})
+    def test_create_tasks_some(self, mock_schedule_tasks, get_old_key_passphrase):
+        ClaButlerDiversityReencryptCommand.chunk_size = 2
+        # Mark the first record as processed so that we can test that diversity_data_reencrypt process does not attempt to
+        # reencrypt this record
+        DiversityDataCheck.objects.create(
+            personal_details_id=self.pd_records_ids[0], action=ACTION.REENCRYPT, status=STATUS.OK
+        )
+
+        def schedule_tasks(tasks, passphrase_old, **kwargs):
```

Comment:
could consider refactoring the schedule_task functions by reusing the same function

Reply:
Updated